content/en/blog/_posts/2022-11-28-kubelet-credential-providers/index.md

```diff
@@ -0,0 +1,128 @@
+the exec plugin communicates with the container registry to dynamically fetch the credentials and returns the credentials in an
+encoded response of the `CredentialProviderResponse` API to the kubelet via stdout.
+
+{{< figure src="kubelet-credential-providers-how-it-works.png" caption="Figure 3: Kubelet credential provider plugin flow" >}}
+
+On receiving credentials from the Kubelet, the plugin can also indicate how long credentials can be cached for, to prevent unecessary
+execution of the plugin by the Kubelet for subsequent image pull requests to the same registry. In cases where the cache duration
+is not specified by the plugin, a default cache duration can be specified by the kubelet (more details below).
+
+```json
+{
+  "apiVersion": "kubelet.k8s.io/v1",
+  "kind": "CredentialProviderResponse",
+  "auth": {
+    "cacheDuration": "6h",
+    "private-registry.io/my-app": {
+      "username": "“user”",
+      "password": "“token12345”"
+    }
+  }
+}
+```
+
+In addition, the plugin can specify the scope in which cached credentials are valid for. This is specified through the `cacheKeyType` field
+in `CredentialProviderResponse`. When the value is `Image`, the kubelet will only use cached credentials for future image pulls that exactly
+match the image of the first request. When the value is `Registry`, the kubelet will use cached credentials for any subsequent image pulls
+that originate from the same registry host, but using different paths (e.g. `gcr.io/foo/bar` and `gcr.io/bar/foo` refer to different images
+from the same registry). And lastly, when the value is `Global`, the kubelet will use returned credentials for all images that match against
+the plugin, including images that can map to different registry hosts (e.g. gcr.io vs k8s.gcr.io). The `cacheKeyType` field is required by plugin
+implementations.
+
+```json
+{
+  "apiVersion": "kubelet.k8s.io/v1",
+  "kind": "CredentialProviderResponse",
+  "auth": {
+    "cacheKeyType": "Registry",
+    "private-registry.io/my-app": {
+      "username": "“user”",
+      "password": "“token12345”"
+    }
+  }
+}
+```
+
+## Using kubelet credential providers
+
+You can configure credential providers by installing the exec plugin(s) into
+a local directory accessible by the kubelet on every node. Then you set two command line arguments for the kubelet:
+* `--image-credential-provider-config`: the path to the credential provider plugin config file.
+* `--image-credential-provider-bin-dir`: the path to the directory where credential provider plugin binaries are located.
+
+The configuration file passed into `--image-credential-provider-config` is read by the kubelet to determine which exec plugins should be invoked for a container image used by a Pod.
+Note that the name of each "provider" must match the name of the binary located in the local directry specified in `--image-credential-provider-bin-dir`, otherwise the Kubelet
```

Comment:
```suggestion
Note that the name of each _provider_ must match the name of the binary located in the local
directory specified in `--image-credential-provider-bin-dir`, otherwise the kubelet
```
“directry” was a typo, and I made other style fixes